File: pytools/data/data.code.lua53.server/ReqRoleCreate.lua
ReqRoleCreate = {}
setmetatable(ReqRoleCreate, {__index = _G})
local _ENV = ReqRoleCreate


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqRoleCreate})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uid = 0
	self.uuid = 0
	self.sid = 0
	self.cid = 0
	self.os = 0
	self.version = 0
	self.uname = 0
	self.source = 0
	self.source_sub = 0
	self.login_time = 0
end


function decode(self, pack)
	self.uid = pack:readU32()
	self.uuid = pack:readU32()
	self.sid = pack:readU16()
	self.cid = pack:readU16()
	self.os = pack:readString()
	self.version = pack:readString()
	self.uname = pack:readString()
	self.source = pack:readString()
	self.source_sub = pack:readString()
	self.login_time = pack:readU32()
	return self
end


function getUid(self)
	return self.uid
end

function getUuid(self)
	return self.uuid
end

function getSid(self)
	return self.sid
end

function getCid(self)
	return self.cid
end

function getOs(self)
	return self.os
end

function getVersion(self)
	return self.version
end

function getUname(self)
	return self.uname
end

function getSource(self)
	return self.source
end

function getSourceSub(self)
	return self.source_sub
end

function getLoginTime(self)
	return self.login_time
end

File: pytools/data/data.code.golang.server/req_role_create.go
package proto

import (
	"packet"
)

type ReqRoleCreate struct {
	uid                      uint32
	uuid                     uint32
	sid                      uint16
	cid                      uint16
	os                       string
	version                  string
	uname                    string
	source                   string
	sourceSub                string
	loginTime                uint32
}

func ReqRoleCreateDecode(pack *packet.Packet) *ReqRoleCreate {
	reqRoleCreate := &ReqRoleCreate{}

	reqRoleCreate.uid = pack.ReadUint32()
	reqRoleCreate.uuid = pack.ReadUint32()
	reqRoleCreate.sid = pack.ReadUint16()
	reqRoleCreate.cid = pack.ReadUint16()
	reqRoleCreate.os = pack.ReadString()
	reqRoleCreate.version = pack.ReadString()
	reqRoleCreate.uname = pack.ReadString()
	reqRoleCreate.source = pack.ReadString()
	reqRoleCreate.sourceSub = pack.ReadString()
	reqRoleCreate.loginTime = pack.ReadUint32()
	return reqRoleCreate
}

func (this *ReqRoleCreate) GetUid() uint32 {
	return this.uid
}

func (this *ReqRoleCreate) GetUuid() uint32 {
	return this.uuid
}

func (this *ReqRoleCreate) GetSid() uint16 {
	return this.sid
}

func (this *ReqRoleCreate) GetCid() uint16 {
	return this.cid
}

func (this *ReqRoleCreate) GetOs() string {
	return this.os
}

func (this *ReqRoleCreate) GetVersion() string {
	return this.version
}

func (this *ReqRoleCreate) GetUname() string {
	return this.uname
}

func (this *ReqRoleCreate) GetSource() string {
	return this.source
}

func (this *ReqRoleCreate) GetSourceSub() string {
	return this.sourceSub
}

func (this *ReqRoleCreate) GetLoginTime() uint32 {
	return this.loginTime
}
